contracts/modules/capital/SafeTracker.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.18;
+
+import "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
+import "../../abstract/MasterAwareV2.sol";
+import "../../interfaces/IPool.sol";
+import "../../interfaces/IPriceFeedOracle.sol";
+import "../../interfaces/ISafeTracker.sol";
+
+contract SafeTracker is ISafeTracker, MasterAwareV2 {
+
+  uint public investedUSDC;
+
+  string public symbol = "NXMIS";
+  string public name = "NXMIS";
+  uint8 public decimals = 18;
+
+  address public immutable safe;
+  IERC20 public immutable aweth;
+  IERC20 public immutable usdc;
+  IERC20 public immutable debtUsdc;
+
+  /* ========== CONSTRUCTOR ========== */
+
+  constructor(address _safe, address _usdc, address _aweth, address _debtUsdc) {
+    safe = _safe;
+    usdc = IERC20(_usdc);
+    aweth = IERC20(_aweth);
+    debtUsdc = IERC20(_debtUsdc);
+  }
+
+  /**
+  * @dev Gets the balance of the safe
+  * @return An uint256 representing the amount of the safe.
+  */
+  function totalSupply() external view returns (uint256) {
+    return _calculateBalance();
+  }
+
+  /**
+  * @dev Gets the balance of the safe
+  * @return An uint256 representing the amount of the safe.
+  */
+  function balanceOf(address account) external view returns (uint256) {
+    if (account != address(pool())) {
+      return 0;
+    }
+    return _calculateBalance();
+  }
+
+  /**
+  * @dev Updates invested USDC
+  */
+  function updateInvestedUSDC(uint _investedUSDC) external {
+    require(msg.sender == safe, "SafeTracker: not safe");
+    investedUSDC = _investedUSDC;
+  }
+
+  /**
+  * @dev emits Transfer event only if it's called by Pool or SwapOperator
+  */
+  function transfer(address, uint256 amount) external returns (bool){
+    if (amount == 0 || msg.sender == internalContracts[uint(ID.P1)] || msg.sender == pool().swapOperator()) {
```

Comment:
you mean amount !== 0?

Reply:
If the amount is 0, it should not revert. You can see Anatol's comment about this.